### PR TITLE
Docs: Implements Documentation to install Cilium on k3s

### LINKED ISSUE #10471: Docs: Document Cilium installation on k3s
Document installation of Cliium on k3s (https://k3s.io)

### PR DESCRIPTION
**Pull request by seanmwinn**
Adds documentation to install Cilium on K3s (https://k3s.io) - a certified distribution of Kubernetes. Previously there was no documentation to create a working K3s sandbox environment running Cilium.

Fixes: #10471

Signed-off-by: Sean Winn <sean@isovalent.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10476)

### CODE REVIEW

**Comment by joestringer** [comment]
test-docs-please

File: Documentation/spelling_wordlist.txt [hunk 1]
```diff
@@ -273,6 +275,7 @@ frontend
 fs
 fwmark
 gRPC
+gc
```
Comment (joestringer) [inline Documentation/spelling_wordlist.txt]: I didn't see where this is used in the document, is this failing on master somewhere?

File: Documentation/spelling_wordlist.txt [hunk 2]
```diff
@@ -153,6 +154,7 @@ Zannoni
 Zhou
 allocator
 allocators
+amd
```
Comment (joestringer) [inline Documentation/spelling_wordlist.txt:157]: Is this avoiding the documentation breakage for using the word `amd64` ?

**Review by joestringer — APPROVED** [review]
This looks nice and easy :-)

(I didn't try it, but I read through)

File: Documentation/spelling_wordlist.txt (hunk 1 above)
Comment (seanmwinn) [reply Documentation/spelling_wordlist.txt]: when I ran the doc tests I was screamed at that this was failing so it was added, but not sure where it came from... so I just updated the commit in my source repo

File: Documentation/spelling_wordlist.txt (hunk 2 above)
Comment (seanmwinn) [reply Documentation/spelling_wordlist.txt:157]: yes!

**Comment by joestringer** [comment]
test-docs-please

**Comment by tgraf** [comment]
test-docs-please

**Review by tgraf — APPROVED** [review]

**Comment by borkmann** [comment]
@seanmwinn noticed that in your commit msg you had written `needs-backport/1.7`. We usually add it to the labels when PR is created.

**Pull request merged**